File: projetos-fenix-para-estudo/t2ti-erp-fenix-servidores-provider-inicio/fontes/backend/java/fenix/src/main/java/com/t2ti/fenix/model/cadastros/Pessoa.java
package com.t2ti.fenix.model.cadastros;

import java.io.Serializable;
import java.util.Set;

import javax.persistence.CascadeType;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.NamedQuery;
import javax.persistence.OneToMany;
import javax.persistence.OneToOne;


/**
 * The persistent class for the pessoa database table.
 * 
 */
@Entity
@NamedQuery(name="Pessoa.findAll", query="SELECT p FROM Pessoa p")
public class Pessoa implements Serializable {
	private static final long serialVersionUID = 1L;

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Integer id;

	private String nome;

	private String tipo;

	private String site;

	private String email;

	private String cliente;

	private String colaborador;

	private String fornecedor;

	private String transportadora;
	
	private String contador;

	@OneToOne(mappedBy = "pessoa", cascade = CascadeType.ALL, orphanRemoval = true)	
	private PessoaJuridica pessoaJuridica;

	@OneToMany(mappedBy = "pessoa", cascade = CascadeType.ALL, orphanRemoval = true)	
	private Set<PessoaContato> listaPessoaContato;
	
	public Pessoa() {
	}

	public Integer getId() {
		return this.id;
	}

	public void setId(Integer id) {
		this.id = id;
	}

	public String getCliente() {
		return this.cliente;
	}

	public void setCliente(String cliente) {
		this.cliente = cliente;
	}

	public String getColaborador() {
		return this.colaborador;
	}

	public void setColaborador(String colaborador) {
		this.colaborador = colaborador;
	}

	public String getEmail() {
		return this.email;
	}

	public void setEmail(String email) {
		this.email = email;
	}

	public String getFornecedor() {
		return this.fornecedor;
	}

	public void setFornecedor(String fornecedor) {
		this.fornecedor = fornecedor;
	}

	public String getNome() {
		return this.nome;
	}

	public void setNome(String nome) {
		this.nome = nome;
	}

	public String getSite() {
		return this.site;
	}

	public void setSite(String site) {
		this.site = site;
	}

	public String getTipo() {
		return this.tipo;
	}

	public void setTipo(String tipo) {
		this.tipo = tipo;
	}

	public String getTransportadora() {
		return this.transportadora;
	}

	public void setTransportadora(String transportadora) {
		this.transportadora = transportadora;
	}

	public String getContador() {
		return contador;
	}

	public void setContador(String contador) {
		this.contador = contador;
	}

	public PessoaJuridica getPessoaJuridica() {
		return pessoaJuridica;
	}

	public void setPessoaJuridica(PessoaJuridica pessoaJuridica) {
		this.pessoaJuridica = pessoaJuridica;
		pessoaJuridica.setPessoa(this);
	}

	public Set<PessoaContato> getListaPessoaContato() {
		return listaPessoaContato;
	}

	public void setListaPessoaContato(Set<PessoaContato> listaPessoaContato) {
		this.listaPessoaContato = listaPessoaContato;
		for (PessoaContato pessoaContato : listaPessoaContato) {
			pessoaContato.setPessoa(this);
		}		
	}
	
}

File: projetos-fenix-para-estudo/t2ti-erp-fenix-banco-agencia-pessoa-completos/fontes/backend/c-sharp/T2TiERPFenix/T2TiERPFenix/Models/Cadastros/Pessoa.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace T2TiERPFenix.Models
{
    [Table("pessoa")]
    public class Pessoa
    {
        [Key]
        [Column("ID")]
        public int Id { get; set; }

        [Column("NOME")]
        public string Nome { get; set; }

        [Column("TIPO")]
        public string Tipo { get; set; }

        [Column("SITE")]
        public string Site { get; set; }

        [Column("EMAIL")]
        public string Email { get; set; }

        [Column("CLIENTE")]
        public string Cliente { get; set; }

        [Column("FORNECEDOR")]
        public string Fornecedor { get; set; }

        [Column("TRANSPORTADORA")]
        public string Transportadora { get; set; }

        [Column("COLABORADOR")]
        public string Colaborador { get; set; }

        [Column("CONTADOR")]
        public string Contador { get; set; }

        [InverseProperty("Pessoa")]
        public PessoaJuridica PessoaJuridica { get; set; }

        [InverseProperty("Pessoa")]
        public PessoaFisica PessoaFisica { get; set; }

        [InverseProperty("Pessoa")]
        public IList<PessoaContato> ListaPessoaContato { get; set; }

        [InverseProperty("Pessoa")]
        public IList<PessoaTelefone> ListaPessoaTelefone { get; set; }

        [InverseProperty("Pessoa")]
        public IList<PessoaEndereco> ListaPessoaEndereco { get; set; }
    }
}
